### DISCUSSION TITLE
Cant build the JSBSimForUnreal.sln

### DISCUSSION THREAD

**Discussion by Thauume** [JSBSim-Team/jsbsim#647] [Support requests]
Hello i need some help i cannot make the Plugin for UE working, building the solution for JSBSimForUnreal fail and i cant fint any solution on Internet

**Comment by CarloBlanch1967** [comment]
I tested it with success on my laptop, what's the problem You encountered?
I followed the  step by step guide.

If I remember well, the first attempt went  wrong.
But second attempt was a success.

**Reply by Thauume** [reply]
I followed step by step on two computer but nothing seems to work :/ , it is > download VS2019 with the right extension (i maybe dont have Toolchain i dont know what is it ) JSBSimForUnreal doesnt compile(or he does when i merge both Master and last release branch like something is missing) download JSBSim.exe setup and compile the uproject but doesnt compile too

**Reply by Thauume** [reply]
Do you have discord ? I would love to get help with pictured etc im a noob but unreal simulation enthusiast add me: Thauume#2128

**Comment by CarloBlanch1967** [comment]
I'm a noob too, for that reason I answered to Your request :-)

I'm using Visual Studio 2019 16.11.19 , (the toolchain is the VS2019 itself) , are You sure You downloaded all the extension needed?
Under "Game Develppement with C++" I checked also "Unreal Engine Installer".
Did You edit the file `BuildConfiguration.xml` ? 
Press Windows Key and digit: `%appdata%`  press on the folder that will appear, then go to "Unreal Engine" and "UnrealBuildTool" and put the file into this folder, edit it if exists.

Did You build JSBSim as a library?
Open VS 2019
Open a project and open the "JSBSimForUnreal.sln"  , make sure that the compiler is in "RELEASE" and "x64" then build it.

**Reply by CarloBlanch1967** [reply]
Just tested,  downloaded the master and I have issues.
I don't remember how I solved the problem, but I  have the working one as backup, and  can compare the two solution to find the difference.

**Reply by Thauume** [reply]
I would like please if u have a correct version

**Comment by gallonmate** [comment]
The repo and plugin are currently being worked on to fix license issues.
For testing try the original, linked below, but keep in mind there's license issues with the original. Then just keep an eye on progress from the master repo.
https://github.com/AlbanBERGERET-Epic/jsbsim/tree/2a5985c2e190ca05f5ecd55b480eed40216dedb6

**Reply by Thauume** [reply]
licence issues ?

**Reply by bcoconni** [reply]
> licence issues ?

@Thauume these are now fixed since PR #644 and PR #650 have been merged.

**Comment by bcoconni** [comment]
@Thauume would you mind checking if the build now succeeds following the updates that have just been introduced (please check out the last commit 4f8d48f33452578df766297cc4b8f54ec0c65788 of the `master` branch).

**Reply by Thauume** [reply]
It does work ! thx

**Comment by SlobodanNinkov** [comment]
One answer, but maybe important for someone new.
JSBSimForUnreal is using ROBOCOPY, and ROBOCOPY fails if you have space " " in your directory name , i.e "Unreal Projects" in your path in front of jsbsim-master.... .
Make sure there is no directories with space in name higher in path of jsbsim-master directory.
